test/interop/fortran/FortranCallChapel/chapelProcs.chpl

```diff
@@ -0,0 +1,29 @@
+var chplInt: int;
+var chplReal: real;
+
+// Note: These exported functions need to be all lowercase
+export proc chpl_library_init_ftn() {
+  extern proc chpl_library_init(argc: c_int, argv: c_ptr(c_ptr(c_char)));
+  var filename = c"fake";
+  chpl_library_init(0, c_ptrTo(filename): c_ptr(c_ptr(c_char)));;
```

Comment:
Note that this doesn't (and probably can't) call the module initialization function, so the globals won't have an initial value unless you call it in the fortran code.  Is it possible to call the module initialization function in the fortran code?  If so, is there a reason we can't do the same for the chpl_library_init function?

Reply:
I'll add a call to the module init function from `chpl_library_init_ftn()`.  I had trouble calling `chpl_library_init()` directly from the Fortran code because of the `char**` argument.  Not saying it can't be done, just that it gave me enough trouble that wrapping it seemed more sane.

Reply:
I'm impressed but pleased that adding the call there works.  I think it's reasonable to do the wrapping you described, at least for now.

On a meta note, what are you thoughts about how this should be handled by the Chapel compiler?  Or would you like to wait until you have more of a handle on the arrays situation? (I'm sure you said something about that in the meeting this morning but I don't remember, sorry :( )

Reply:
I think the compiler could generate a function like `chpl_library_init_ftn()` that calls both `chpl_library_init()` and the module's init function so that both are handled in a single call. That seems like it would be a nice simplification.

Reply:
Fortran handles multiple files similar to how C does.  Compile each file into a .o and then link them.

That would make sense and be useful.  If we were to generate the module `ChapelInterface` as it appears in testCallChapel.f90, but into its own file, that could be `use`d by a Fortran program to get access to the functions without needing to do anything more.

Reply:
Awesome!  Would you mind opening an issue for that?

Reply:
Issue #11364

Reply:
Thanks!